### PR TITLE
:bug: fix for h-omer#8

### PR DESCRIPTION
**Pull request by tschn**
check to see if relationship end node matches before deleting

### CODE REVIEW

**Comment by mfalcier** [comment]
Hi! And thanks for your contribution! May I ask you if you could also provide some tests for what you've implemented? Thanks!

**Comment by tschn** [comment]
sure, I will work on those next week, thanks!

**Comment by mfalcier** [comment]
Thank you so much!

**Comment by tschn** [comment]
test added, please let me know if you want additional testing

File: src/main/java/org/homer/versioner/core/procedure/RelationshipProcedure.java [hunk 1]
```diff
@@ -28,6 +30,36 @@
...
+    public Stream<RelationshipOutput> relationshipsCreate(
+            @Name("entitySource") Node entitySource,
+            @Name("entityDestinations") List<Node> entityDestinations,
+            @Name(value = "relProps", defaultValue = "[{}]") List<Map<String, Object>> relProps,
+            @Name(value = "date", defaultValue = "null") LocalDateTime date) {
+
+        Optional<Node> sourceCurrentState = createNewSourceState(entitySource, defaultToNow(date));
+        isEntityOrThrowException(entitySource);
+
+        return entityDestinations.size() == relProps.size() ? sourceCurrentState.map(node -> zip(entityDestinations, relProps).stream().map((item) -> {
```
Comment (albertodeste) [inline src/main/java/org/homer/versioner/core/procedure/RelationshipProcedure.java:44]: Is it possible to use explicit "if" condition? It is more readable.

File: src/main/java/org/homer/versioner/core/Utility.java [hunk 2]
```diff
@@ -222,7 +221,20 @@ public static LocalDateTime convertEpochToLocalDateTime(Long epochDateTime) {
         return Instant.ofEpochMilli(epochDateTime).atZone(ZoneId.systemDefault()).toLocalDateTime();
     }
 
-    public static Boolean isSystemType(String type) {
+    public static <A, B> List<org.apache.commons.lang3.tuple.Pair<A, B>> zip(List<A> listA, List<B> listB) {
+        List<Pair<A, B>> pairList = new LinkedList<>();
```
[the remark this answers is not in the document]
Comment (tschn) [reply src/main/java/org/homer/versioner/core/Utility.java:230]: please see latest commit

File: src/main/java/org/homer/versioner/core/procedure/RelationshipProcedure.java [hunk 3]
```diff
@@ -28,6 +30,36 @@
...
+            final Node destinationNode = item.getLeft();
```
[the remark this answers is not in the document]
Comment (tschn) [reply src/main/java/org/homer/versioner/core/procedure/RelationshipProcedure.java:48]: please see latest commit

File: src/main/java/org/homer/versioner/core/procedure/RelationshipProcedure.java (hunk 1 above)
Comment (tschn) [reply src/main/java/org/homer/versioner/core/procedure/RelationshipProcedure.java:47]: please see latest commit

**Comment by mfalcier** [comment]
Hi @tschn , can you please explain the new procedure you've added for multiple relationship delete?
Can we also add it to the documentation next?

**Comment by mfalcier** [comment]
@tschn is there any chance you're going to explain your changes? we would like to integrate those, but it would be nice to hear something from you!

**Comment by tschn** [comment]
please see latest commit for updated documentation, the plural procedures were added to allow updating more than one relationship for a given state for 'batch' style updates to the graph

**Pull request merged**